web/ex_admin/errors_helper.ex

```diff
@@ -0,0 +1,86 @@
+defmodule ExAdmin.ErrorsHelper do
+  @moduledoc """
+    The primary purpose of this module is to take nested changeset errors created
+    by many_to_many and has many through relationships and change them into a format
+    that the forms can use to get the error message from the field name.
+
+    Changes sets such as:
+    #Ecto.Changeset<action: nil,
+      changes: %{phone_numbers: [#Ecto.Changeset<action: :update, changes: %{},
+         errors: [], data: #ContactDemo.PhoneNumber<>, valid?: true>,
+          #Ecto.Changeset<action: :update, changes: %{},
+            errors: [number: {"can't be blank", []}], data: #ContactDemo.PhoneNumber<>,
+            valid?: false>,
+          #Ecto.Changeset<action: :insert, changes: %{label: "Primary Phone"},
+            errors: [number: {"can't be blank", []}], data: #ContactDemo.PhoneNumber<>,
+            valid?: false>]},
+      errors: [], data: #ContactDemo.Contact<>, valid?: false>
+
+    need to be walked and each of the error messages needs to be flattened into its
+    appropriately namespaced verison.
+
+    To do this we need both the changeset and the schema used to generate the changeset.
+    This is required because we need to look at the schema to properly create the neccesary
+    form field names. For example, many_to_many association have attributes appended to the
+    field name so that we know it is a many to many field.
+  """
+  def create_errors(changeset, schema) do
+    assoc_prefixes = create_prefix_map(schema)
+    Enum.filter(List.flatten(flatten_errors(changeset, assoc_prefixes)), fn(x) -> x != nil end)
+  end
+
+  defp flatten_errors(errors_array, assoc_prefixes, prefix \\ nil)
+  defp flatten_errors(%Ecto.Changeset{changes: changes, errors: errors}, assoc_prefixes, prefix) do
+    if(length(errors) > 0 and prefix != nil) do
```

Comment:
Like earlier comments, you cannot use a variable set inside an if case. Actually, probably best to use 2 function clauses for flatten_errors here.
```
defp flatten_errors(%Ecto.Changeset{changes: changes, errors: errors}, assoc_prefixes, prefix) when errors == [] or is_nil(prefix) do
  errors ++ flatten_errors(changes, assoc_prefixes, prefix)
end
defp flatten_errors(%Ecto.Changeset{changes: changes, errors: errors}, assoc_prefixes, prefix)  do
  Enum.map(errors, fn({k, v}) -> {concat_atoms(prefix, k), v} end) ++ 
    flatten_errors(changes, assoc_prefixes, prefix)
end
```